presto-native-execution/presto_cpp/main/SystemConnector.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "presto_cpp/main/SystemSplit.h"
+#include "presto_cpp/main/types/PrestoToVeloxConnector.h"
+
+#include "velox/connectors/Connector.h"
+
+namespace facebook::presto {
+
+class TaskManager;
+
+class SystemColumnHandle : public velox::connector::ColumnHandle {
+ public:
+  explicit SystemColumnHandle(const std::string& name) : name_(name) {}
+
+  const std::string& name() const {
+    return name_;
+  }
+
+ private:
+  const std::string name_;
+};
+
+class SystemTableHandle : public velox::connector::ConnectorTableHandle {
+ public:
+  explicit SystemTableHandle(
+      std::string connectorId,
+      std::string schemaName,
+      std::string tableName);
+
+  ~SystemTableHandle() override {}
+
+  std::string toString() const override;
+
+  const std::string& schemaName() {
+    return schemaName_;
+  }
+
+  const std::string& tableName() {
+    return tableName_;
+  }
+
+  const velox::RowTypePtr& taskSchema();
+
+ private:
+  const std::string schemaName_;
+  const std::string tableName_;
+};
+
+class SystemDataSource : public velox::connector::DataSource {
+ public:
+  SystemDataSource(
+      const velox::RowTypePtr& outputType,
+      const std::shared_ptr<velox::connector::ConnectorTableHandle>&
+          tableHandle,
+      const std::unordered_map<
+          std::string,
+          std::shared_ptr<velox::connector::ColumnHandle>>& columnHandles,
+      const TaskManager* taskManager,
+      velox::memory::MemoryPool* pool);
+
+  void addSplit(
+      std::shared_ptr<velox::connector::ConnectorSplit> split) override;
+
+  void addDynamicFilter(
+      velox::column_index_t /*outputChannel*/,
+      const std::shared_ptr<velox::common::Filter>& /*filter*/) override {
+    VELOX_NYI("Dynamic filters not supported by SystemConnector.");
+  }
+
+  std::optional<velox::RowVectorPtr> next(
+      uint64_t size,
+      velox::ContinueFuture& future) override;
+
+  uint64_t getCompletedRows() override {
+    return completedRows_;
+  }
+
+  uint64_t getCompletedBytes() override {
+    return completedBytes_;
+  }
+
+  std::unordered_map<std::string, velox::RuntimeCounter> runtimeStats()
+      override {
+    // TODO: Which stats do we want to expose here?
```

Comment:
nit: Unless somebody comes up with some useful statistics to expose we should probably drop the comment